### PR TITLE
with no_grad in evaluation step.

### PR DESCRIPTION
**Pull request by stephenroller**
Agents generally call `self.model.eval()` in their `eval_step`, so we're accurately computing dropout etc. However, without explicitly disabling the autograd infrastructure, we're wasting memory and compute. This is particularly noticeable when running beam search.

It makes sense to just have a torch nograd around the call to eval step. This will disable it, by default for everyone and probably prevent some headaches later. Users can manually reenable grads using

```python
with torch.enable_grad():
    pass
```

### CODE REVIEW

**Comment by stephenroller** [comment]
Actually seems to shave ~30s off of travis, but that could be noise.
